core/arxiv/submission/process/checks/tex_produced/__init__.py

```diff
@@ -0,0 +1,354 @@
+"""arXiv TeX produced routines.
+
+This package checks a submission file object to determine whether it was generated
+from TeX. arXiv requires the TeX source, when available, to make long-term maintenance
+of papers easier. Single-file submission determined to be generated by TeX are rejected.
+
+The tex-produced check current works on both streams and files for PDFs and just files
+for Postscript files. The original code only worked with files. In NG we pass around streams
+of file content which would require writing out in order to user the file-based check.
+
+We now convert the non-seekable stream to an in-memory object and use the PyPDF2 package
+to extract info and fonts information from a PDF.
+
+There are then a specific set of information we look for in the PDF info and certain
+fonts that indicate the content was generated by some form of TeX engine.
+
+"""
+import io
+import re
+import subprocess
+import os
+import tempfile
+
+from typing import IO, List
+from arxiv.base import logging  # type: ignore
+
+from PyPDF2 import PdfFileReader, utils
+
+logger = logging.getLogger(__name__)
+
+verbose = 0
+
+
+def get_filtered_pdf_info_from_stream(stream: IO[bytes]) -> List[str]:
+    """
+    Returns select set of values from pdfinfo output for specified file.
+
+    Parameters
+    ----------
+    stream: IO[bytes
+        The stream to run pdfinfo on.
+
+    Returns
+    -------
+    List containing one entry for each selected line from pdfinfo output.
+
+    """
+    info_list = []
+
+    try:
+        pdf = PdfFileReader(stream)
+        information = pdf.getDocumentInfo()
+
+        if information.creator is not None:
+            info_list.append(bytes('Creator:        ' + information.creator, 'utf-8'))
+
+        if information.title is not None:
+            info_list.append(bytes('Title:          ' + information.title, 'utf-8'))
+
+        if information.producer is not None:
+            info_list.append(bytes('Producer:       ' + information.producer, 'utf-8'))
+        return info_list
+    except utils.PdfReadError as ex:
+        logger.error(f"get_filtered_pdf_info_from_stream: Can't read PDF: Error: {ex}")
+        raise ex
+    except Exception as ex:
+        logger.error(f"Error: Unknown {ex}")
+        raise ex
+
+    return info_list
+
+
+def get_filtered_pdf_info_from_file(filepath: str) -> List[str]:
+    """
+    Returns select set of values from pdfinfo output for specified file.
+
+    Parameters
+    ----------
+    filepath : str
+        The file to run pdfinfo on.
+
+    Returns
+    -------
+    List containing one entry for each selected line from pdfinfo output.
+
+    """
+    info_list = []
+
+    info = subprocess.run(["pdfinfo", f"{filepath}"], stderr=subprocess.PIPE,
```

Comment:
The original module dealt with files so I left the original calls in just in case this package migrates elsewhere. At one point I was also concerned about whether the PyPDF2 package would work as well as the popplar PDF tools. I can remove the file-based routines.

Reply:
I'd say get rid of them if we are not using them.

Reply:
I've eliminates the file-based routines that use the install pdfinfo/pdffonts commands.